01-tree-specification.bs

```diff
@@ -134,19 +136,63 @@ This report also explains how clients MAY implement support for extracting conte
+First we introduce a couple of symbols:
+ * `N` is the set of all named graphs used in the current page, including the default graph.
+ * `Nb` is the set of blank nodes in `N`, excluding the default graph.
+ * `f` is the first focus node of a member we are extracting found in a `<yourcollection> tree:member ?f` triple.
+ * `F` is the set of all member focus nodes matching the `<yourcollection> tree:member ?f` pattern.
+
+A client extracting members MUST iterate over all terms `f` in `F`.
+`f` can be either a blank node, a named node, or a triple term.
+If `f` is a triple term, the triple itself is the member.
+If `f` is a literal an error MUST be returned.
+
+In the case it is an IRI or a blank node, the client MUST look for further triples as follows:
+ 1. A set of named graphs that need to be ignored `I` MUST be created. These are named graphs that will be explicitely used to package triples to be entirely part of `M`. `I` is the union of `Nb` and `F`.
+ 2. For each `g` in `N \ ( Nb ∪ F)`, resolve the quad pattern `GRAPH g { f ?p ?o }` and add those quads to `M`.
+ 3. For every object `o` that is a result of `?o` in step 2 that is a blank node,
+     - repeat step 2 with that blank node as `f`.
+	 - solve the quad pattern `GRAPH o { ?s ?p ?o }` and add all quads to `M`.
+ 4. resolve the quad pattern `GRAPH f { ?s ?p ?o }` and add all quads to `M`.
+ 5. If `M` is still an empty set of quads, dereference `f` and perform the algorithm on that response again.
+
+Issue: In step 5: should we perform the algorithm again on the quads in the response, or should we just add all quads found in the response and thus use the triples in the file as a package?
+
+## The Generalized Member Extraction Algorithm ## {#generalized-member-extraction-algorithm}
+
+Note: As the Shape Topologies algorithm requires more advanced processing, we discourage publishers from relying on it as it will have a negative effect on client performance. On top of that, while the TREE MEA is straightforward to implement, the generalized MEA is more tedious for developers, which results in a large support for the TREE MEA among client tooling, while the Shape Topologies algorithm may get omitted. It does however has functionalities that are not supported by the more simple TREE MEA.
+
+When on the root `tree:Node`, `tree:shapeTopology` has been set to true, the algorithm in the [Shape Topology report](https://w3id.org/tree/specification/shape-topologies) MUST be used to extract the set of quads as intended by the server using the `tree:shape`.
+This is an extension of the TREE Member Extraction Algorithm in which a SHACL shape can indicate:
+ 1. Partially out of band members
+ 2. Based on the shape, include or exclude a more specific set of quads. It does however not perform full validation of the members.
+
+A client MAY also provide a configuration flag to still extract it using the Shape Topology algorithm in case the `tree:shapeTopology` property has not been set.
```

Comment:
```suggestion
A client MAY still choose to perform extraction using the Shape Topology algorithm (e.g. indicated by a configuration flag) in case the `tree:shapeTopology` property has not been set to true.
```

Reply:
I see the issue you have with the sentence - rewrote it myself in the next commit: was not entirely satisfied with your rewrite as it’s not the client that is chosing, but the end-user.